Isosurfacing_3/include/CGAL/Isosurfacing_3/internal/Dual_contouring_internal.h

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) 2022 INRIA Sophia-Antipolis (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Daniel Zint
+//                 Julian Stahl
+
+#ifndef CGAL_DUAL_CONTOURING_3_INTERNAL_DUAL_CONTOURING_3_H
+#define CGAL_DUAL_CONTOURING_3_INTERNAL_DUAL_CONTOURING_3_H
+
+#include <CGAL/Bbox_3.h>
+#include <CGAL/Isosurfacing_3/internal/Tables.h>
+#include <CGAL/Origin.h>
+
+#include <array>
+#include <eigen3/Eigen/SVD>
+#include <map>
+#include <mutex>
```

Comment:
@sloriot why do we have `<CGAL/mutex.h>`?

Reply:
historical reasons, now it is probably useless (except if one wants to use the CGAL macros defined).

Reply:
That header was before the migration to C++14. At the time when CGAL was supporting older compilers (without C++11), `<CGAL/mutex.h>` and `<CGAL/thread.h>` was providing an implementation either from Boost or C++11's STL.

Reply:
So we might clean up.